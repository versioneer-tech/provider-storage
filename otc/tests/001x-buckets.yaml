---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-joe
spec:
  bucketAccessGrants:
  - bucketName: s-joe
    grantedAt: "2025-09-29T10:05:00Z"
    grantee: jeff
    permission: ReadWrite
  bucketAccessRequests:
  - bucketName: s-jeff-shared
    reason: Need access
    requestedAt: "2025-09-29T10:00:00Z"
  buckets:
  - bucketName: s-joe
    discoverable: true
  principal: joe
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-joe, bucket-s-joe, bucketpolicy-s-joe,
      and 4 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-joe
  labels:
    crossplane.io/composite: s-joe
  name: access-key-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-joe
      spec:
        forProvider:
          type: AK
          userId: "111"
        providerConfigRef:
          name: provider-otc
        writeConnectionSecretToRef:
          name: joe
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucket-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: Bucket
      metadata:
        name: s-joe
      spec:
        forProvider:
          bucket: s-joe
          region: eu-nl
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucketpolicy-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucketpolicy-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-s-joe
      spec:
        forProvider:
          bucket: s-joe
          policy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "ListBucket"
                  ],
                  "Resource": [
                    "s-joe"
                  ],
                  "Principal": {
                    "ID": [
                      "domain/222:user/111"
                    ]
                  }
                },
                {
                  "Effect": "Allow",
                  "Action": [
                    "GetObject",
                    "PutObject",
                    "DeleteObject"
                  ],
                  "Resource": [
                    "s-joe/*"
                  ],
                  "Principal": {
                    "ID": [
                      "domain/222:user/111"
                    ]
                  }
                }
              ]
            }
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-bucketpolicy-s-joe
  labels:
    crossplane.io/composite: s-joe
  name: observe-bucketpolicy-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-s-joe
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-credential-joe
  labels:
    crossplane.io/composite: s-joe
  name: observe-credential-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-joe
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-user-jeff
  labels:
    crossplane.io/composite: s-joe
  name: observe-user-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jeff
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-joe
  labels:
    crossplane.io/composite: s-joe
  name: user-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: joe
      spec:
        forProvider:
          enabled: true
          name: joe
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
