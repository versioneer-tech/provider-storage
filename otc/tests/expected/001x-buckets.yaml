---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-joe
spec:
  bucketAccessGrants:
  - bucketName: b-joe
    grantee: jeff
    permission: ReadWrite
  bucketAccessRequests:
  - bucketName: b-jeff-shared
    reason: Need access
  buckets:
  - bucketName: b-joe
    discoverable: true
  principal: joe
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-joe, bucket-b-joe, bucketpolicy-b-joe,
      and 4 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-joe
  labels:
    crossplane.io/composite: s-joe
  name: access-key-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-joe
      spec:
        forProvider:
          type: AK
          userId: "111"
        providerConfigRef:
          name: provider-otc
        writeConnectionSecretToRef:
          name: joe
          namespace: default
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucket-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: Bucket
      metadata:
        name: b-joe
      spec:
        forProvider:
          bucket: b-joe
          region: eu-nl
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucketpolicy-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: bucketpolicy-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-b-joe
      spec:
        forProvider:
          bucket: b-joe
          policy: |-
            {
              "Version": "2012-10-17",
              "Statement": [
                {
                  "Effect": "Allow",
                  "Action": [
                    "ListBucket"
                  ],
                  "Resource": [
                    "b-joe"
                  ],
                  "Principal": {
                    "ID": [
                      "domain/222:user/111"
                    ]
                  }
                },
                {
                  "Effect": "Allow",
                  "Action": [
                    "GetObject",
                    "PutObject",
                    "DeleteObject"
                  ],
                  "Resource": [
                    "b-joe/*"
                  ],
                  "Principal": {
                    "ID": [
                      "domain/222:user/111"
                    ]
                  }
                }
              ]
            }
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-bucketpolicy-b-joe
  labels:
    crossplane.io/composite: s-joe
  name: observe-bucketpolicy-b-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: obs.opentelekomcloud.crossplane.io/v1alpha1
      kind: BucketPolicy
      metadata:
        name: bucketpolicy-b-joe
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-credential-joe
  labels:
    crossplane.io/composite: s-joe
  name: observe-credential-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: CredentialV3
      metadata:
        name: aksk-joe
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-user-jeff
  labels:
    crossplane.io/composite: s-joe
  name: observe-user-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: jeff
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-joe
  labels:
    crossplane.io/composite: s-joe
  name: user-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-joe
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: identity.opentelekomcloud.crossplane.io/v1alpha1
      kind: UserV3
      metadata:
        name: joe
      spec:
        forProvider:
          enabled: true
          name: joe
        providerConfigRef:
          name: provider-otc
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
